### PR TITLE
Foundations Installing Node.js lesson and Arrays and Loops lesson: Provide descriptive link text #27681 

### PR DESCRIPTION
**Pull request by kudamndau**
## Because
This PR updates  descriptive link text to links in the TOP Foundations' JavaScript Basics: Installing Node.js lesson and JavaScript Basics: Arrays and Loops lesson. This makes the link text succinctly descriptive and improves accessibility

## This PR

-  Replaces original link text in 1 link in the TOP Foundations' JavaScript Basics: Installing Node.js lesson and 6 links in the TOP Foundations' JavaScript Basics: Arrays and Loops lesson.
- Uses descriptive link texts from the TOP Foundations: Provide accessible, descriptive links Google spreadsheet.

## Issue
Related to #27681

## Pull Request Requirements

-   [x ] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x ] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x ] The `Because` section summarizes the reason for this PR
-   [x ] The `This PR` section has a bullet point list describing the changes in this PR
-   [x ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [ ] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: foundations/javascript_basics/arrays_and_loops.md [hunk 1]
```diff
@@ -17,16 +17,16 @@ This section contains a general overview of topics that you will learn in this l
 
 Strings and numbers may be our building blocks, but as your scripts get more complex, you're going to need a way to deal with large quantities of them. Luckily, JavaScript has a couple of data types that are used for just that. An Array is an ordered collection of items (Strings, numbers, or other things).
 
-1. [This tutorial](https://www.w3schools.com/js/js_arrays.asp) is a great introduction.
-1. [This article](https://www.w3schools.com/js/js_array_methods.asp) covers some of the most useful built-in array methods. These fundamentals are something you'll use every day, so don't rush too much and miss out!
-1. [Web Dev Simplified video](https://www.youtube.com/watch?v=7W4pQQ20nJg) explains an overview of arrays in JavaScript in about 6 minutes.
+1. Go through [JavaScript Arrays from W3Schools](https://www.w3schools.com/js/js_arrays.asp).
+1. [JavaScript Array Methods](https://www.w3schools.com/js/js_array_methods.asp) covers some of the most useful built-in array methods. covers some of the most useful built-in array methods. These fundamentals are something you'll use every day, so don't rush too much and miss out!
```
Comment (MaoShizhong) [inline foundations/javascript_basics/arrays_and_loops.md:21]:
```suggestion
1. [JavaScript Array Methods](https://www.w3schools.com/js/js_array_methods.asp) covers some of the most useful built-in array methods. These fundamentals are something you'll use every day, so don't rush too much and miss out!
```

File: foundations/javascript_basics/arrays_and_loops.md [hunk 2]
```diff
@@ -17,16 +17,16 @@ This section contains a general overview of topics that you will learn in this l
...
+1. This [JavaScript Arrays crash course](https://www.youtube.com/watch?v=7W4pQQ20nJg) explains an overview explains an overview of arrays in JavaScript in about 6 minutes.
```
Comment (MaoShizhong) [inline foundations/javascript_basics/arrays_and_loops.md:22]:
```suggestion
1. This [JavaScript Arrays crash course](https://www.youtube.com/watch?v=7W4pQQ20nJg) explains an overview of arrays in JavaScript in about 6 minutes.
```

File: foundations/javascript_basics/arrays_and_loops.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ We will teach you the art of actually writing these tests later in the course. T
 
 <div class="lesson-content__panel" markdown="1">
 
-1. Go to our [JavaScript exercises repository](https://github.com/TheOdinProject/javascript-exercises) and review the [repository's README](https://github.com/TheOdinProject/javascript-exercises#readme) to set up your local environment. Once you have cloned the repository and installed Jest, review each README file prior to completing the following exercises in order:
+1. Go and review the [README of our `javascript-exercises` repository](https://github.com/TheOdinProject/javascript-exercises#readme) to set up your local environment. to set up your local environment. Once you have cloned the repository and installed Jest, review each README file prior to completing the following exercises in order:
```
Comment (MaoShizhong) [inline foundations/javascript_basics/arrays_and_loops.md:43]:
```suggestion
1. Go and review the [README of our `javascript-exercises` repository](https://github.com/TheOdinProject/javascript-exercises#readme) to set up your local environment. Once you have cloned the repository and installed Jest, review each README file prior to completing the following exercises in order:
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
@kudamndau Thanks for making these changes!

Just a couple of them I think you accidentally pasted without replacing some of the required text, so there's a little text duplication for some of them.

I have highlighted which bits need removing below - after that, we'll be good to merge :rocket:

File: foundations/javascript_basics/arrays_and_loops.md (hunk 2 above)
Comment (kudamndau) [reply foundations/javascript_basics/arrays_and_loops.md]: @MaoShizhong I have corrected the typos, please check and let me know if everything is in order.

**Review by MaoShizhong — APPROVED** [review]
All good now! Thank you for making these changes :rocket:

**Pull request merged**